docs-src/design-decisions.md

```diff
@@ -0,0 +1,44 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Enables using Service API on a management cluster designed to provision and
+  manage other clusters and infrastructure needed by other clusters.
+* Easier to provide simple and secure defaults (only allow Gateways and Routes
```

Comment:
I think "simple and secure" is very biased wording.  It would allow a different form of access control that what non-namespaced classes allow, but "different" != "simpler" or "more secure"

Reply:
Good feedback, I've updated this to:
>Enables a more restrictive (and potentially more secure) default for where a
  GatewayClass could be used. This could mean only allowing Gateways and Routes
  in the same namespace by default.